src/main/java/com/splunk/kafka/connect/SplunkSinkTask.java

```diff
@@ -250,39 +326,99 @@ private Event createHecEventFrom(final SinkRecord record) {
+                log.error("event does not follow correct HEC pre-formatted format", record.toString());
+                event = createHECEventNonFormatted(record);
+            }
+        } else {
+            event = createHECEventNonFormatted(record);
         }
 
-        Map<String, String> metas = connectorConfig.topicMetas.get(record.topic());
-        if (metas != null) {
-            event.setIndex(metas.get(SplunkSinkConnectorConfig.INDEX));
-            event.setSourcetype(metas.get(SplunkSinkConnectorConfig.SOURCETYPE));
-            event.setSource(metas.get(SplunkSinkConnectorConfig.SOURCE));
-            event.addFields(connectorConfig.enrichments);
+        if(connectorConfig.headerSupport) {
+            addHeaders(event, record);
         }
 
         if (connectorConfig.trackData) {
-            // for data loss, latency tracking
             Map<String, String> trackMetas = new HashMap<>();
             trackMetas.put("kafka_offset", String.valueOf(record.kafkaOffset()));
             trackMetas.put("kafka_timestamp", String.valueOf(record.timestamp()));
             trackMetas.put("kafka_topic", record.topic());
             trackMetas.put("kafka_partition", String.valueOf(record.kafkaPartition()));
             event.addFields(trackMetas);
         }
-
         event.validate();
 
         return event;
     }
 
+    private Event addHeaders(Event event, SinkRecord record) {
+        Headers headers = record.headers();
+        if(headers.isEmpty() &&  connectorConfig.headerCustom.isEmpty()) {
+            return event;
+        }
+
+        if (headers.lastWithName(connectorConfig.headerIndex) != null) {
+        //log.debug("splunk_index header detected, value is: " + headers.lastWithName(connectorConfig.headerIndex).value().toString());
+            event.setIndex(headers.lastWithName(connectorConfig.headerIndex).value().toString());
+        }
+        if (headers.lastWithName(connectorConfig.headerHost) != null) {
+            //log.debug("splunk_host header detected, value is: " + headers.lastWithName(connectorConfig.headerHost).value().toString());
+            event.setHost(headers.lastWithName(connectorConfig.headerHost).value().toString());
+        }
+        if (headers.lastWithName(connectorConfig.headerSource) != null) {
+            //log.debug("splunk_source header detected, value is: " + headers.lastWithName(connectorConfig.headerSource).value().toString());
+            event.setSource(headers.lastWithName(connectorConfig.headerSource).value().toString());
+        }
+        if (headers.lastWithName(connectorConfig.headerSourcetype) != null) {
+            //log.debug("splunk_sourcetype header detected, value is: " + headers.lastWithName(connectorConfig.headerSourcetype).value().toString());
+            event.setSourcetype(headers.lastWithName(connectorConfig.headerSourcetype).value().toString());
+        }
+
+        // Custom headers are configured with a comma separated list passed in configuration
```

Comment:
Let's make sure, these are header keys, not headers and we don't support key containing ","

Reply:
We will support a comma separated list of custom headers provided to the configuration to index as extra fields